iset.mm

```diff
@@ -535,14 +612,19 @@ The Metamath (Metamath-exe) program Proof Assistant requires proofs to
 $)
 
   $( If ` ph ` is a wff, so is ` -. ph ` or "not ` ph ` ."  Part of the
-     recursive definition of a wff (well-formed formula).  Traditionally, Greek
+     recursive definition of a wff (well-formed formula).  In intuitionistic
```

Comment:
I wanted to put a sentence at that place to contrast with analogous sentences at the same place in set.mm and nf.mm.

What about "In intuitionistic logic, a formula need not be provable nor refutable." ? (or "...need not be either provable or refutable" ?) And maybe add "...contrary to the case of classical logic"). ? (should it be simply "contrary to classical logic" ?)

Reply:
In classical logic there are formulas which are not provable nor refutable (per Goedel's incompleteness theorems).

I take it the sentence you would like to contrast with is "In classical logic (which is our logic), a wff is interpreted as either true or false." How you interpret truth values in intuitionistic logic is something which a lot of people have tried to explain but the usual treatise launches into BHK interpretations (example: section 3.2 of [Bauer]) which seems too far afield for current purposes.

Oddly enough, set.mm follows the sentence I quote above with "So if 𝜑 is true, then ¬ 𝜑 is false; if 𝜑 is false, then ¬ 𝜑 is true." which are provable in iset.mm, at least in all the ways I can think of to formalize them. Starting with `𝜑 -> ¬ ¬ 𝜑` and ` ¬ 𝜑 -> ¬ 𝜑 ` but we could also point to things like https://us.metamath.org/ileuni/notfal.html and https://us.metamath.org/ileuni/nottru.html or something related like ` ( ( ph <-> T. ) -> ( -. ph <-> F. ) ) ` which I'm not sure we have in exactly that form but which is provable in iset.mm.

Reply:
> In classical logic there are formulas which are not provable nor refutable (per Goedel's incompleteness theorems).

Since we're talking of classical logic, I'd prefer to quote Gödel's completeness theorem ! (But you're right that my proposed formulation is not good.)

> I take it the sentence you would like to contrast with is "In classical logic (which is our logic), a wff is interpreted as either true or false." How you interpret truth values in intuitionistic logic is something which a lot of people have tried to explain but the usual treatise launches into BHK interpretations (example: section 3.2 of [Bauer]) which seems too far afield for current purposes.

> Oddly enough, set.mm follows the sentence I quote above with "So if 𝜑 is true, then ¬ 𝜑 is false; if 𝜑 is false, then ¬ 𝜑 is true." which are provable in iset.mm

Good catch ! set.mm misses the point. We should write something like "Therefore, if -. phi is false, then phi is true, see ~ notnotr (a property which need not hold in intuitionistic logic)."

Reply:
I merely removed that sentence in my latest commit.